File: 3764-maximum-sum-with-at-most-k-elements/3764-maximum-sum-with-at-most-k-elements.js
/**
 * @param {number[][]} grid
 * @param {number[]} limits
 * @param {number} k
 * @return {number}
 */
var maxSum = function (grid, limits, k) {
    let list = []
    for (let row of grid)
        row.sort((a, b) => {
            return b - a;
        });
    for (let i = 0; i < limits.length; i++) {
        for (let j = 0; j < limits[i]; j++)
            list.push(grid[i][j]);
    }
    list.sort((a, b) => {
        return b - a;
    });

    let sum = 0;
    for (let i = 0; i < k; i++)
        sum += list[i];

    return sum;
};

File: 3764-maximum-sum-with-at-most-k-elements/3764-maximum-sum-with-at-most-k-elements.cs
public class Solution {
    public long MaxSum(int[][] grid, int[] limits, int k) {
        var list = new List<int>();
        foreach (var row in grid)
            Array.Sort<int>(row, new Comparison<int>(
                (i1, i2) => i2.CompareTo(i1)));
        
        for (int i = 0; i < limits.Length; i++) {
            for (int j = 0; j < limits[i]; j++)
                list.Add(grid[i][j]);
        }
        list.Sort((a, b) => b - a);
        long sum = 0;

        for (int i = 0; i < k; i++)
            sum += list[i];
            
        return sum;
    }
}